server/src/test/java/org/elasticsearch/search/geo/GeoPointShapeQueryTests.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.geo;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.geo.builders.*;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.locationtech.jts.geom.Coordinate;
+
+import static org.elasticsearch.action.support.WriteRequest.RefreshPolicy.IMMEDIATE;
+import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
+
+public class GeoPointShapeQueryTests extends GeoQueryTests {
+
+    @Override
+    protected XContentBuilder createMapping() throws Exception {
+        XContentBuilder xcb = XContentFactory.jsonBuilder().startObject()
+            .startObject("properties").startObject("location")
+            .field("type", "geo_point")
+            .endObject().endObject().endObject();
+
+        return xcb;
+    }
+
```

Comment:
I think this test should be in GeoQueryTests? I guess the idea of the subclass is that all test with points should be in that class so they are called with geo_point fields and geo_shape field?

Reply:
Thanks @iverase 

I've pushed all tests that may be candidates for both geo shape and geo point up to the parent class, reordered geo shape so these are more easily visible and created test stubs in geo point for the expected tests

Reply:
@djptek Thanks for reworking this butI think the is still not the idea.I would propose to leave GeoShapeQueryTest as it is and only move out the test that is indexing points. This test can be in the abstract class with will be implemented by two classes, one generating a `geo_point` field and the other generated a `geo_shape` field. Does it make sense?

Reply:
@iverase thanks for clarifying, I've refactored leaving 2 tests in the abstract: 

- testNullShape() which is theoretically applicable to both
- testIndexPointsFilterRectangle() as discussed offline which is the test of interest

The testIndexPointsFilterRectangle() method is overridden in the subclasses since:

- geo shape requires a random mapping to maintain current test coverage
- geo point lacks the concept of random mapping therefore uses the default; also this test is comment out since the field mapper has not yet been built so it will fail to build

Also refactored the mapping helping method function names as discussed

Thanks for your support and feedback